pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -382,17 +385,17 @@ public String getReplicatorPrefix() {
             switch (producer.getAccessMode()) {
             case Shared:
                 if (hasExclusiveProducer || !waitingExclusiveProducers.isEmpty()) {
-                    return FutureUtil.failedFuture(new ProducerBusyException(
-                            "Topic has an existing exclusive producer: " + producers.keys().nextElement()));
+                    return FutureUtil.failedFuture(
+                            new ProducerFencedException("Topic has an existing exclusive producer: " + exclusiveProducer));
```

Comment:
A shared producer will not get fenced, instead it will get a busy error

Reply:
good catch